### ISSUE TITLE
igxGrid row selection

### ISSUE THREAD

**Issue by Lipata** [IgniteUI/igniteui-angular#812]
Implement row selection in igxGrid.

### Research 
- [x] [Research document](https://infragisticsinc297.sharepoint.com/:w:/r/Groups/ProdDev/_layouts/15/doc.aspx?sourcedoc=%7B63740d66-6ec6-4b84-bb54-ea9cf813902a%7D&action=edit&uid=%7B63740D66-6EC6-4B84-BB54-EA9CF813902A%7D&ListItemId=3882&ListId=%7BED8CDA43-BB2D-4B0E-8ABD-5C7F4F99DC28%7D&odsp=1&env=prod)
- [x] [Technical specification](https://github.com/IgniteUI/igniteui-angular/wiki/Row-Selection-(Grid-feature))
- [x] POC

### Development
- [x] POC polish
- [x] Create row selection
- [x] Create header checkbox selection
- [x] Change cell selection
- [x] Filtering integration
- [x] Paging integration
- [x] Sorting integration
- [x] Pinning integration
- [x] Aria Support
- [x] Keyboard Navigation
- [x] API according to spec
- [x] Styling & UI implementation
- [x] Unit tests

### Documentation
- [x] ReadMe
- [x] DocFX
- [x] Sample creation
- [x] Changelog update

### Links
- development branch (https://github.com/IgniteUI/igniteui-angular/tree/nalipiev/row-selection)
- specification (https://github.com/IgniteUI/igniteui-angular/wiki/Row-Selection-(Grid-feature))

**Comment by SlavUI** [comment]
@SDinev  and @StefanIvanov  Hi guys we need a design for this one if we have it please include it in the spec.

**Comment by Eralmidia** [comment]
"the only mode of selection will be multiple"

Does this mean it's not possible to force selection of a single row? This seems like a huge limitation if it's the case. What is the reasoning behind this? At least in our solution it's vastly more common to only select a single row at a time. So we will basically have to monitor selection and deselect manually if we want to force single row selection?

**Comment by SlavUI** [comment]
@Eralmidia The igx-grid should support selection of one or more rows. I don't think there will be a limitation on selecting a single row.

**Comment by Lipata** [comment]
Hi, @Eralmidia. This specification is still initial one, and any matter is a matter of discussion. Thank you for your question and for sure it will be taken in account.
The idea of that is to have an easy configuration of the row selection feature and in the same time to cover most of the user cases. Of course our API will support events and methods that will easily allow you to turn the default multiple selection into single one. We will propose event on which you will be able to call a method that will deselect all the current selected rows.
The basic idea behind this is to keep the things simple - this will also lead to a fewer future problems of the control.

**Comment by Eralmidia** [comment]
I see. Keeping it simple is of course a good thing, and I guess it's not a big deal whether its through an event or an option, as long as this is well documented and doesn't increase complexity in development. I imagine this will be a very common use case. I'm sure we have more than 100 ig grids in our solution, and that more than 90% of them uses single row selection.

**Comment by Lipata** [comment]
@kdinev can you check the feature API in the specifications? Thank you!

**Comment by Lipata** [comment]
@kdinev I've updated the specification. I've added grid methods that will work with row identifiers that are either primaryKey or rowData. I left the grid methods that will still accept row components as parameters. The idea is to use them to mimic the user row selection that is happening on visible rows. We still have getRowByIndex method that returns a row component. I'm not sure which is the best, but if we still have the getRowByIndex method, then we may leave the selection methods that work with row component.
